src/components/PurchaseOrder/PODetails/PODetailsForm.js

```diff
@@ -298,46 +316,68 @@ class PODetailsForm extends Component {
               type="checkbox"
             />
           </Col>
+          <Col
+            xs={6}
+            lg={3}
+          >
+            <FieldOrderType />
+          </Col>
           <Col
             xs={6}
             lg={3}
           >
             <Field
-              component={TextField}
+              component={Selection}
+              dataOptions={ownerOptions}
               fullWidth
-              label={<FormattedMessage id="ui-orders.orderDetails.billTo" />}
-              name="billTo"
+              label={<FormattedMessage id="ui-orders.poLine.owner" />}
+              name="owner"
             />
           </Col>
+        </Row>
+        <Row>
           <Col
             xs={6}
             lg={3}
           >
             <Field
-              component={TextField}
+              component={Selection}
+              dataOptions={addEmptyOption(addressesOptions)}
               fullWidth
-              label={<FormattedMessage id="ui-orders.orderDetails.shipTo" />}
-              name="shipTo"
+              label={<FormattedMessage id="ui-orders.orderDetails.billTo" />}
+              name="billTo"
+              onChange={(e, value) => this.onChangeField(value, 'billTo')}
             />
           </Col>
-        </Row>
-        <Row>
           <Col
             xs={6}
             lg={3}
           >
-            <FieldOrderType />
+            <KeyValue
+              label={<FormattedMessage id="ui-orders.orderDetails.address" />}
+              value={addressBillTo}
+            />
           </Col>
           <Col
             xs={6}
             lg={3}
           >
             <Field
               component={Selection}
-              dataOptions={ownerOptions}
+              dataOptions={addEmptyOption(addressesOptions)}
               fullWidth
-              label={<FormattedMessage id="ui-orders.poLine.owner" />}
-              name="owner"
+              label={<FormattedMessage id="ui-orders.orderDetails.shipTo" />}
+              name="shipTo"
+              onChange={(e, value) => this.onChangeField(value, 'shipTo')}
```

Comment:
this shouldn't be in the final code. These values should go to the back-end

Reply:
Fixed